programs/mmm/src/instructions/cnft/metadata_args.rs

```diff
@@ -0,0 +1,63 @@
+use anchor_lang::{prelude::*, AnchorDeserialize, AnchorSerialize};
+
+// Below types are copied from mpl bubblegum crate so
+// IDL will automatically include them
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Debug, Eq, PartialEq, PartialOrd, Hash)]
+pub enum TokenStandard {
+    NonFungible,
+    FungibleAsset,
+    Fungible,
+    NonFungibleEdition,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Debug, Eq, PartialEq)]
+pub struct Collection {
+    pub verified: bool,
+    pub key: Pubkey,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Debug, Eq, PartialEq, PartialOrd, Hash)]
+pub enum UseMethod {
+    Burn,
+    Multiple,
+    Single,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Debug, Eq, PartialEq)]
+pub struct Uses {
+    pub use_method: UseMethod,
+    pub remaining: u64,
+    pub total: u64,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Debug, Eq, PartialEq, PartialOrd, Hash)]
+pub enum TokenProgramVersion {
+    Original,
+    Token2022,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Debug, Eq, PartialEq)]
+pub struct Creator {
+    pub address: Pubkey,
+    pub verified: bool,
+    /// The percentage share.
+    ///
+    /// The value is a percentage, not basis points.
+    pub share: u8,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone)]
+pub struct MetadataArgs {
```

Comment:
look into if we can pass in serialized data

Reply:
tried with serialize and deserialize, the deserilized data has some unknown issue that I couldn't figure out in short time (it fails allowlist check, means the deserialization had some issue). will keep the metadata args as it is for now to unblock this PR